Makefile

```diff
@@ -81,7 +81,7 @@ ENVSUBST_VER := v2.0.0-20210730161058-179042472c46
 ENVSUBST_BIN := envsubst
 ENVSUBST := $(TOOLS_BIN_DIR)/$(ENVSUBST_BIN)-$(ENVSUBST_VER)
 
-GOLANGCI_LINT_VER := v1.50.0
+GOLANGCI_LINT_VER := v1.50.1
```

Comment:
related to my comment in https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/3032#issuecomment-1385836005, I wonder if there's any way for us to automate these makefile dependencies... they tend to be forgotten and fall behind if we don't remember to update them and make PRs like this one

Reply:
Dependabot doesn't know how to find such dependencies in Makefiles. We could perhaps add them to `go.mod` in order to clue in dependabot, but then we'd be maintaining versions in two places unless we're clever somehow...

Having said that, we should merge this change to see if it fixes the linter timeout we've been seeing often. Otherwise, I'll create a PR to bump up the timeout.